library/psa_crypto.c

```diff
@@ -22,462 +22,451 @@
 
 #if defined(MBEDTLS_PSA_CRYPTO_C)
 
-#if defined(MBEDTLS_PSA_CRYPTO_CONFIG)
-#include "check_crypto_config.h"
-#endif
-
-#include "psa/crypto.h"
-
-#include "psa_crypto_cipher.h"
-#include "psa_crypto_core.h"
-#include "psa_crypto_invasive.h"
-#include "psa_crypto_driver_wrappers.h"
-#include "psa_crypto_ecp.h"
-#include "psa_crypto_hash.h"
-#include "psa_crypto_mac.h"
-#include "psa_crypto_rsa.h"
-#include "psa_crypto_ecp.h"
-#if defined(MBEDTLS_PSA_CRYPTO_SE_C)
-#include "psa_crypto_se.h"
-#endif
-#include "psa_crypto_slot_management.h"
+#    if defined(MBEDTLS_PSA_CRYPTO_CONFIG)
```

Comment:
I was not familiar with this but it seems quite good to spot at a glance the level of nesting of pre-processor guards. But this first indentation because of "#if defined(MBEDTLS_PSA_CRYPTO_C)" is a bit of a pity and does not bring much I would say. Would be there a way to avoid it ?

Reply:
I'm not sure I understand.
Are you happy with preprocessor directives indentation or not?
If yes, are you suggesting that some of them should be indented and others not, regardless of nesting level?
Please clarify :)

Reply:
Having skimmed a bit more through the reformatted code preprocessor directives indentations make the code less readable to me. Thus I vote against but no veto, just my vote.

Reply:
Alright, then it's:
* Against indented PP directives: 1 vote
* For indented PP directives: 1 vote (me :P)

Waiting for other opinions.
Also - there is also an option to indent with the '#', so it would be:
<code>&nbsp;&nbsp;&nbsp;&nbsp;#devine MACRO</code> instead of 
<code>#&nbsp;&nbsp;&nbsp;&nbsp;devine MACRO</code>

Reply:
I have a weak preference for intended directives, with the `#` in column 1. I'd prefer them to have a shorter indentation than braced code: 2 instead of 4, or 1 instead of 2.

Like Ronald, I dislike having the whole-file conditional count as an indentation level. In a header file, which has the de facto standard `#ifndef HEADER_NAME_H` guard, that guard doesn't count:
```
#ifndef FOO_H
#define FOO_H
#if SOME_CONDITION
#    if NESTED_CONDITION
#        define WIBBLE
#    endif // NESTED_CONDITION
#endif // SOME_CONDITION
#endif // foo.h
```
But in our C files, which have a `MBEDTLS_MODULE_C` guard, that does count:
```
#if defined(MBEDTLS_FOO_C)
#    include "mbedlts/foo.h"
#    if CONDITION
#        define WOBBLE
#    endif // CONDITION
#endif // foo.c
```
Ronald's request was, can we have this instead in .c files?
```
#if defined(MBEDTLS_FOO_C)
#include "mbedlts/foo.h"
#if CONDITION
#    define WOBBLE
#endif // CONDITION
int f(void)
{
    return 42;
}
#endif // foo.c
```

And I'd prefer less indent on preprocessor directives:
```
#if defined(MBEDTLS_FOO_C)
#include "mbedlts/foo.h"
#if CONDITION
#  define WOBBLE
#endif // CONDITION
int f(void)
{
    return 42;
}
#endif // foo.c
```

Reply:
I don't think it's possible to have some predefined level of nesting from which indentation would be applied. We could experiment with //clang-format on/off, but this would severly impact code readability.

Custom indent for preprocessor directovies comes with clang-format-13, the option is named `PPIndentWidth`.
clang-13 is not available in apt repo for ubuntu 18.04 (which I am currently using), so that's why I am using clang-12.
Further going into details on version differences should be preceded by agreeing on which clang-format version should the CI use.